<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>749</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Kalk</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QPushButton{
	border: 2px solid black;
	color:black;
}
QPushButton:hover{
	background:yellow;
}
QPushButton:pressed{
	border: 2px solid red;
	background:blue;
	color:white;
}
QLabel {
	color:black;
}
QSpinBox{
	border: 2px solid black;
 	color: black; 
}
QSpinBox:hover{
	border: 2px solid black;
 	background:yellow;
}
QSpinBox:focus{
	border: 2px solid red;
	background: yellow;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QScrollArea" name="scrollCreateTreeArea">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>70</y>
      <width>341</width>
      <height>251</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
    </property>
    <widget class="QWidget" name="scrollContentCreateTreeArea">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>339</width>
       <height>249</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>330</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string>Result Display : </string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>601</width>
      <height>23</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="maximumSize">
        <size>
         <width>131</width>
         <height>29</height>
        </size>
       </property>
       <property name="text">
        <string>Track operation : </string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="expressionArea">
       <property name="styleSheet">
        <string notr="true">border: 2px solid green;
</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>461</width>
      <height>23</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_4">
       <property name="maximumSize">
        <size>
         <width>100</width>
         <height>29</height>
        </size>
       </property>
       <property name="text">
        <string>Messages : </string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="errorArea">
       <property name="styleSheet">
        <string notr="true">border: 2px solid orange;
color: red;</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>30</y>
      <width>155</width>
      <height>341</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="operationsBox">
     <property name="spacing">
      <number>6</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="UndoTreeButton">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;undo tree with chosen index&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Undo Tree</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="delTreeSpinBox">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;choose tree's index to delete&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="clearAllTreesButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Clear all trees&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Clear All Trees</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="UndoOperationButton">
       <property name="minimumSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Undo last operation&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Undo Operation</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ClearAllOperationButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Clear all shown operations&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Clear All Operations</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="executeButton">
       <property name="minimumSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Calculate Result&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>=</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SumMiddleButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;TernaryTrees-Only&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>SumMiddle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SumValuesButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sum nodes values&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>SumValues</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="BlackHeightButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;RedBlackTrees-Only&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>BlackHeight</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SumRedNodesButton">
       <property name="text">
        <string>SumRedNodes</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SumBlackNodesButton">
       <property name="text">
        <string>SumBlackNodes</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="BinaryRotationButton">
       <property name="text">
        <string>BinaryRotation</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="IntersectionButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Intersect Trees&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Intersection</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SumButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sum Trees&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Sum</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>111</width>
      <height>167</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="styleSheet">
        <string notr="true">QLabel {
	color:black;
}</string>
       </property>
       <property name="text">
        <string>Create Tree :</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="createBinaryTree">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;create Binary Tree&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Binary</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="createRBBinaryTree">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;create Binary-RB Tree&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>BInary-RB</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="createTernaryTree">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;create Ternary Tree&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>Ternary</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="styleSheet">
        <string notr="true">QLabel {
	color:black;
}</string>
       </property>
       <property name="text">
        <string>Set Height : </string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="TreeHeightBox">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Set Tree Height&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>4</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>111</width>
      <height>271</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="treeSelectorArea"/>
   </widget>
   <widget class="QScrollArea" name="ResultArea">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>370</y>
      <width>341</width>
      <height>251</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>339</width>
       <height>249</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>410</y>
      <width>111</width>
      <height>271</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="treeSelectOp"/>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>316</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string>Shown Tree :</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>380</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string>Select Operand : </string>
    </property>
   </widget>
   <widget class="QLabel" name="ResultExpression">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>640</y>
      <width>411</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <zorder>label_3</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>verticalLayoutWidget</zorder>
   <zorder>ResultArea</zorder>
   <zorder>verticalLayoutWidget_2</zorder>
   <zorder>label_6</zorder>
   <zorder>label_7</zorder>
   <zorder>ResultExpression</zorder>
   <zorder>scrollCreateTreeArea</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>640</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
